examples/cosmos/ics20/ics20.qnt

```diff
@@ -325,6 +350,25 @@ module ics20Test {
     chainStates' = chainStates.set(chain, newChainState)
   }
 
+  /// Receive a timeout on `chain`:
+  /// - Pick an arbitrary, previously unreceived timeout on `chain`.
+  /// - Call the `onTimeoutPacket` callback on it.
+  /// - Update the chain state to record the timeout as received.
+  action receiveTimeout(chain: str): bool = all {
+    // (1) Compute updated chain state:
+    // non-deterministically pick an acknowledgement
+    val chainState = chainStates.get(chain)
+    nondet timedoutPacket = chainState.inTimeouts.oneOf()
+    // call the `onTimeout` callback and
+    val timedoutChainState = onTimeoutPacket(chainState, timedoutPacket)
+    // remove `timedoutPacket` from incoming timeouts
+    val newInTimeouts = chainState.inTimeouts.exclude(Set(timedoutPacket))
+    val newChainState = timedoutChainState.with("inTimeouts", newInTimeouts)
```

Comment:
Is it possible in IBC to both timeout a packet and acknowledge it? I do not see why this should not be possible.

Reply:
We discussed synchronously. Igor sees now ;-)